scripts/dev/clang-format.sh

```diff
@@ -93,10 +93,10 @@ for p in "$@"; do
     if [[ -d "$p" ]]; then
         :
     elif [[ -f "$p" ]]; then
-        if [[ "$p" =~ .*\.[ch] ]]; then
+        if [[ "$p" =~ .*\.([ch]|h\.in) ]]; then
             :
         else
-            err "File '$p' is not a C source file"
+            err "File '$p' is not a C file"
```

Comment:
Seems like this would generate a lot of noise...

Reply:
This just checks command line args, so if you run `./scripts/dev/clang-format.sh  doc/blah.md`  it will fail with a helpful error message -- before running the formatter.